docs/source/example_npe_model.md

```diff
@@ -0,0 +1,127 @@
+--------------------------------
+
+To generate an ASD dataset we can run the same command as in the previous tutorial.
+
+```
+dingo_generate_asd_dataset --settings_file examples/npe_model/asd_dataset_settings_fiducial.yaml --data_dir $TRAIN_DIR/asd_dataset_folder -out_name $TRAIN_DIR/asds_O1_fiducial.hdf5
+```
+
+However, this time, during training we will need two sets of ASDs. The first one will be
+fixed during the initial training -- this is the fiducial dataset generated above.
+This dataset will contain only  a single ASD. The second ASDDataset will contain many
+ASDs and is used during the fine tuning stage. To generate this second dataset run
+
+```
+dingo_generate_asd_dataset --settings_file $(pwd)/examples/npe_model/asd_dataset_settings.yaml --data_dir $TRAIN_DIR/asd_dataset_folder -out_name $TRAIN_DIR/asds_O1.hdf5
+```
+
+We can see that in `examples/npe_model/asd_dataset_settings.yaml` the `num_psds_max`
+attribute is set to `0` indicating that all possible ASDs will be downloaded. If you want to 
+decrease this, make sure that there are enough ASDs in the training set to represent 
+any possible data the dingo network will see. Typically this should be at least 1000,
+but of course more is better. 
+
+
+Step 3 Training the network
+---------------------------
+
+Now we are ready for training. The command is analogous to the previous tutorial 
+but the settings are increased to production values. To run the training do
+
+```
+sed -i 's+/path/to/waveform_dataset.hdf5+'"$TRAIN_DIR"'/waveform_dataset.hdf5+g' examples/npe_model/train_settings.yaml
+sed -i 's+/path/to/asds_fiducial.hdf5+'"$TRAIN_DIR"'/asd_dataset_folder/asds_O1_fiducial.hdf5+g' examples/npe_model/train_settings.yaml
+sed -i 's+/path/to/asds.hdf5+'"$TRAIN_DIR"'/asd_dataset_folder/asds_O1.hdf5+g' examples/npe_model/train_settings.yaml
+dingo_train --settings_file examples/npe_model/train_settings.yaml --train_dir $TRAIN_DIR
+```
+
+```{tip}
+If running on a machine with multiple GPUs make sure to specify the GPU by running 
+export `CUDA_VISIBILE_DEVICES=GPU_NUM` before running `dingo_train`
+```
+
+The main difference from the toy example in the network architecture is the size of the embedding
+network which is described in `model.embedding_net_kwargs.hidden_dims` and the
+number of neural spline flow transforms described in
+`model.nsf_kwargs.num_flow_steps`. These increase the depth of the network and the 
+number/size of the layers in the embedding network. 
+
+Also notice there are now two training stages `stage_0` and `stage_1`. In `stage_0` a fixed ASD is used and the reduced basis layer
+is frozen. Then in `stage_1` all ASDs are used and the reduced basis layer is unfrozen. 
+
+The main difference in the local settings is that the `device` is set to `CUDA`.
+Note if you have multiple GPUs on the machine, you can select which GPU to use
+by running 
+
+```{important}
+It is recommended to have at least 40 GB of GPU memory on the device. 
```

Comment:
What should a user do if they have < 40 GB of GPU RAM? (Reduce the batch size, or network size.)

Reply:
https://github.com/dingo-gw/dingo/pull/173/commits/1963348ee56b5af159d1f31d2ddab2dc7652faa0